backend/src/main/java/gov/cdc/usds/simplereport/api/queue/QueueMutationResolver.java

```diff
@@ -34,22 +36,27 @@ public QueueMutationResolver(TestOrderService tos, PersonService ps) {
       _ps = ps;
     }
 
-    public void addTestResult(String deviceID, String result, String patientID, String dateTested) {
+    public void addTestResult(String deviceID, String result, String patientID, String dateTested) throws ParseException {
+      DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
```

Comment:
In the event that we don't offload this to the plugin, we should make this a library function that relies on a constant—recreating the formatter every call is silly.

Reply:
I believe that parsers in java.time are thread safe (and this was not true in the before-time). Citation: [DateTimeFormatter javadocs](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/format/DateTimeFormatter.html).